src/main/java/io/fabric8/jenkins/openshiftsync/ConfigMapWatcher.java

```diff
@@ -190,6 +198,43 @@ private boolean containsSlave(ConfigMap configMap) {
                 podTemplate = xStream2.fromXML(entry.getValue());
 
                 if( podTemplate instanceof PodTemplate ) {
+                    PodTemplate pt = (PodTemplate)podTemplate;
+                    
+                    String image = pt.getImage();
+                    try {
+                        // if requested via special prefix, convert this images entry field, if not already fully qualified, as if
+                        // it were an IST
+                        if (image.startsWith(SPECIAL_IST_PREFIX)) {
+                            image = image.substring(SPECIAL_IST_PREFIX_IDX);
+                            if (!OpenShiftUtils.hasRegistry(image)) {
+                                // if no registry it should have one and only one '/' at most
+                                boolean hasNamespace = hasOneAndOnlyOneWithSomethingAfter(image, "/");
+                                boolean hasTag  = hasOneAndOnlyOneWithSomethingAfter(image, ":");
+                                String namespace = getAuthenticatedOpenShiftClient().getNamespace();
+                                String isName = image;
+                                String newImage = null;
+                                if (hasNamespace) {
+                                    String[] parts = image.split("/");
+                                    namespace = parts[0];
+                                    isName = parts[1];
+                                }
+                                if (hasTag) {
+                                    ImageStreamTag ist = getAuthenticatedOpenShiftClient().imageStreamTags().inNamespace(namespace).withName(isName).get();
+                                    if (ist != null && ist.getImage() != null && ist.getImage().getDockerImageReference() != null && ist.getImage().getDockerImageReference().length() > 0) {
+                                        newImage = ist.getImage().getDockerImageReference();
+                                        logger.fine(String.format("Converting image ref %s as an imagestreamtag %s to fully qualified image %s", image, isName, newImage));
+                                    }
+                                }
+                                
+                                if (newImage != null) {
+                                    pt.getContainers().get(0).setImage(newImage);
+                                }
+                            }
```

Comment:
there are a few cases in here where we silently ignore their "image" value:

1) if they provided a value like imagestreamtag:docker.io/centos:foo
2) if they provided a value with no tag (imagestreamtag:ruby)

i think if we get to newImage==null we should at least log the imagevalue we read in and print a warning.

similarly, it's probably worth logging cases where it was a parseable imagestreamtag but we couldn't find/resolve the reference.

and if it's a totally invalid value (e.g. has two slashes, two colons, no colons, etc) we should log that.